doc/advopt.txt

```diff
@@ -67,6 +67,11 @@ Advanced options:
   --clib:LIBNAME            link an additional C library
                             (you should omit platform-specific extensions)
   --project                 document the whole project (doc2)
+  --docRoot:path            nim doc --docRoot:/foo --project --outdir:docs /foo/sub/main.nim
+                            generates: docs/sub/main.html
+                            if path == @pkg, will use nimble file enclosing dir
+                            if path == @path, will use first matching dir in --path
+                            if these are nonexistant, will use project path
```

Comment:
Why do we need this? Can't we be smart about the project path and do without this?

Reply:
short answer: I've now added a new smart default for `--docRoot`:
`--docRoot` with no arg now implies `--docRoot:@default` which picks the best match among @path, @pkg (and if these have no match, defaults to project path)

most users should simply use `--docRoot` with no arg.

## long answer
before introducing `@default`, the way I wrote it accomodated all scenarios, which all happen in practice:
* user has a nimble package and no --path => use @pkg; eg: `compiler` nimble package (fallsback to project path if no nimble file found)
* user has (1 or more) --path => use @path, eg: `stdlib` which has `--path:$nim/lib/pure`, `--path:$nim/lib/pure/concurrency` etc  (fallsback to project path if no enclosing --path found)
* user has no nimble package and no path => use provided absolute dir

In all likelihood, @default will be more common for user docs, as I noted in the docs

now that I have @default, I'd be ok to remove @pkg and @path (but keep option for user provided absolute dir) but maybe it's best to keep these as option